mlos_bench/mlos_bench/config/schemas/tunables/tunable-params-schema.json

```diff
@@ -35,6 +35,12 @@
                     },
                     "minItems": 1,
                     "uniqueItems": true
+                },
+                "weights": {
```

Comment:
Proposal:
Change
```json
"specials": ["flat", "list", "of", "special", "values"],
"weights": ["list", "must", "align", "gives", "no", "context"]
```
to
```json
"specials": ["flat", "list", "implicitly", "uniform", "as", "before"],
"specials": {
  "val1": {
    "description": "docs",
    "meta": {
       "extra": "info",
    },
    "weight": 1  // the default if omitted
  },
},
```

The latter seems more powerful and is easily converted into other forms if need be.

Reply:
Same should be possible for categoricals btw.

Reply:
One challenge I see with this version would be weights of the special values relative to the range, but that could be handled via a separate field: for instance
```jsonc
"range_bias": 0.9,
```

Which gives 90% chance of choosing a range value instead of a special value.

Reply:
Also, there are several other prioritzation policies we could explore here:
- always sample special values first (there are likely to be few of them), then switch over to the range
- have some sort of decay function where we sample specials with high probability initially, and decrease over time
- reverse that

Reply:
That's all good, but I don't want to make this PR too big. My plan is the following:
1. Add weights to special and categorical values and pass this data all the way to the ConfigSpace and the optimizer (this PR)
2. Run some experiments and see how it works in practice, e.g., for SQL Server QO
3. Add a way to specify distributions (beta, log uniform, etc) in another PR and see what else is missing. This can be an opportunity to refactor this PR and see if we need a different syntax for weights and other stuff (e.g., grouping each special value with weight and metadata)

Who knows, maybe we will want to keep weights and distributions separately from the basic tunables' data?